From 79f136afdff7c03745e6f26f511157e674a7e7f2 Mon Sep 17 00:00:00 2001
From: Brett Neumeier <brett@neumeier.us>
Date: Tue, 18 Aug 2009 19:43:23 -0500
Subject: [PATCH 1/2] remove reference to missing file

---
 libc/manual/libc.texinfo |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/libc/manual/libc.texinfo b/libc/manual/libc.texinfo
index 85f2f20..f05a53f 100644
--- a/libc/manual/libc.texinfo
+++ b/libc/manual/libc.texinfo
@@ -11,7 +11,6 @@
 @end direntry
 @include dir-add.texi
 
-@include pkgvers.texi
 
 @c This tells texinfo.tex to use the real section titles in xrefs in
 @c place of the node name, when no section title is explicitly given.
-- 
1.6.3.3

