diff -r -N -u eglibc-2.10.1-orig/manual/libc.texinfo eglibc-2.10.1/manual/libc.texinfo
--- eglibc-2.10.1-orig/manual/libc.texinfo	2009-08-24 14:59:08.000000000 -0400
+++ eglibc-2.10.1/manual/libc.texinfo	2009-08-24 14:59:45.000000000 -0400
@@ -11,7 +11,6 @@
 @end direntry
 @include dir-add.texi
 
-@include pkgvers.texi
 
 @c This tells texinfo.tex to use the real section titles in xrefs in
 @c place of the node name, when no section title is explicitly given.
